File: bench/02-list-pipeline.py
from functools import reduce

def sum_even_squares(n: int) -> int:
    return reduce(lambda acc, x: acc + x,
                  [x * x for x in range(1, n + 1) if x % 2 == 0], 0)

def top_n(ns: list[int], n: int) -> list[int]:
    return sorted(ns, reverse=True)[:n]

def tag_list(tags: list[str], prefix: str) -> int:
    return len([t for t in tags if t.startswith(prefix)])

File: bench/02-list-pipeline.java
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.LongStream;

final class ListPipeline {
  public static long sumEvenSquares(long n) {
    return LongStream.range(1, n + 1).filter(x -> x % 2 == 0).map(x -> x * x).sum();
  }

  public static List<Long> topN(List<Long> ns, int n) {
    return ns.stream().sorted((a, b) -> Long.compare(b, a)).limit(n).collect(Collectors.toList());
  }

  public static long tagList(List<String> tags, String prefix) {
    return tags.stream().filter(t -> t.startsWith(prefix)).count();
  }
}
